openfe/protocols/openmm_utils/system_creation.py

```diff
@@ -145,8 +145,8 @@ def get_omm_modeller(protein_comp: Optional[ProteinComponent],
       Protein Component, if it exists.
     solvent_comp : Optional[ProteinCompoinent]
       Solvent Component, if it exists.
-    small_mols : list[SmallMoleculeComponents]
-      List of SmallMoleculeComponents to add.
+    small_mols : list
+      List of OpenFF Molecule to add.
     omm_forcefield : app.ForceField
       ForceField object for system.
     solvent_settings : SolvationSettings
```

Comment:
this is very ugly currently as it's taking a mixture of gufe objects and openff.

The proteincomponent goes straight to openmm rather than having an intermediate openff object representation, so it's not clear there's an easy way to have all objects in the same "type".  system_creation is pretty internal so it's probably not something that is that bad though

Reply:
Oof ok, this is messier than I thought - I completely missed the fact that we're now keying a dictionary that tracks the residues of the system with a mixture of components & offmols.

That dictionary is then going everywhere in the rest of the protocol which escapes the scope of just this method, that's going to cause someone using our tooling (like Ivan) a massive headache.